File: samples/server/petstore/finch/src/main/scala/Pet.scala
package io.swagger.petstore

import argonaut.Argonaut._
import argonaut.CodecJson
//import Category
//import Tag

/**
 * A pet for sale in the pet store
 * @param id 
 * @param category 
 * @param name 
 * @param photoUrls 
 * @param tags 
 * @param status pet status in the store
 */
case class Pet(id: Option[Long],
                category: Option[Category],
                name: String,
                photoUrls: List[String],
                tags: Option[List[Tag]],
                status: Option[String]
                )

object Pet {
  /**
   * Creates the codec for converting Pet from and to JSON.
   */
  implicit val PetCodec: CodecJson[Pet] = casecodec6(Pet.apply, Pet.unapply)("id", "category", "name", "photoUrls", "tags", "status")
}

File: samples/client/petstore/flutter/swagger/lib/model/pet.dart
part of swagger.api;

class Pet {
  
  int id = null;
  

  Category category = null;
  

  String name = null;
  

  List<String> photoUrls = [];
  

  List<Tag> tags = [];
  
/* pet status in the store */
  String status = null;
  //enum statusEnum {  available,  pending,  sold,  };
  Pet();

  @override
  String toString() {
    return 'Pet[id=$id, category=$category, name=$name, photoUrls=$photoUrls, tags=$tags, status=$status, ]';
  }

  Pet.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    id = json['id'];
    category = new Category.fromJson(json['category']);
    name = json['name'];
    photoUrls = json['photoUrls'];
    tags = new Tag.fromJson(json['tags']);
    status = json['status'];
  }

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'category': category == null ? null : category.toMap(),
      'name': name,
      'photoUrls': photoUrls,
      'tags': tags == null ? null : tags.toMap(),
      'status': status
     };
  }
}

